lib/puppet/defaults.rb

```diff
@@ -1711,6 +1711,12 @@ def self.initialize_default_settings!(settings)
       can be guaranteed to support this format, but it will be used for all
       classes that support it.",
     },
+    :allow_pson_serialization => {
+      :default    => true,
+      :type       => :boolean,
+      :desc       => "Whether when unable to serialize to JSON or other formats, Puppet falls back to to PSON.
+        This option is useful to prevent the loss of data because rich data cannot be serialized via PSON.",
```

Comment:
Might be good to call out this setting affects both puppetserver's configuration management service responses https://www.puppet.com/docs/puppet/latest/server/http_api_index.html#configuration-management-services and when the agent saves its cached catalog. Also looks like the docs page needs some updates, so I'll file a ticket on that.